include/itkRLEImageScanlineIterator.h

```diff
@@ -102,6 +102,13 @@ class ImageScanlineIterator<RLEImage<TPixel, VImageDimension, CounterType>>
     return *this;
   }
 };
+
+// Deduction guide for class template argument deduction (CTAD).
+template <typename TPixel, unsigned int VImageDimension, typename CounterType>
+ImageScanlineIterator(SmartPointer<RLEImage<TPixel, VImageDimension, CounterType>>,
+                      const typename RLEImage<TPixel, VImageDimension, CounterType>::RegionType &)
+  ->ImageScanlineIterator<RLEImage<TPixel, VImageDimension, CounterType>>;
+
```

Comment:
Shouldn't raw pointer CTAD be added here as well?

Reply:
It seems that this is only a workaround to a compiler bug. Especially because GCC and MSVC don't seem to need it. In that case, if it works, it works! But of course, if clang also needs a deduction guide for raw pointers, you may add one. 

@dzenanz [Nitpick] when you say "CTAD", I think you mean _deduction guide_! (CTAD = class template argument deduction).

Reply:
I tried the raw pointer for the non-Const, was not compatible.